src/validation/extensions/maxar/MaxarValidatorCommon.ts

```diff
@@ -0,0 +1,246 @@
+import { defined } from "3d-tiles-tools";
+
+import { ValidationContext } from "../../ValidationContext";
+import { BasicValidator } from "../../BasicValidator";
+
+import { JsonValidationIssues } from "../../../issues/JsonValidationIssues";
+
+/**
+ * A class for validating common elements that appear in both
+ * `MAXAR_image_ortho` and `MAXAR_grid` extensions.
+ *
+ * @internal
+ */
+export class MaxarValidatorCommon {
+  /**
+   * Validates the spatial reference system (srs) object
+   *
+   * @param path - The path for ValidationIssue instances
+   * @param srs - The srs object to validate
+   * @param context - The ValidationContext that any issues will be added to
+   * @returns Whether the object was valid
+   */
+  static validateSrs(
+    path: string,
+    srs: any,
+    context: ValidationContext
+  ): boolean {
+    // Make sure that the given value is an object
+    if (!BasicValidator.validateObject(path, "srs", srs, context)) {
+      return false;
+    }
+
+    let result = true;
+
+    // Validate required properties
+    const requiredProperties = [
+      "referenceSystem",
+      "epoch",
+      "coordinateSystem",
+      "elevation",
+    ];
+    for (const property of requiredProperties) {
+      if (!defined(srs[property])) {
+        const propertyPath = path + "/" + property;
+        const message = `The '${property}' property is required`;
+        const issue = JsonValidationIssues.PROPERTY_MISSING(
+          propertyPath,
+          message
+        );
+        context.addIssue(issue);
+        result = false;
+      }
+    }
+
+    // Validate referenceSystem enum
+    const referenceSystem = srs.referenceSystem;
+    if (defined(referenceSystem)) {
+      const referenceSystemPath = path + "/referenceSystem";
+      const validReferenceSystemValues = ["WGS84-G1762", "ITRF2008"];
+      if (
+        !BasicValidator.validateEnum(
+          referenceSystemPath,
+          "referenceSystem",
+          referenceSystem,
+          validReferenceSystemValues,
+          context
+        )
+      ) {
+        result = false;
+      }
+    }
+
+    // Validate epoch pattern
+    const epoch = srs.epoch;
+    if (defined(epoch)) {
+      const epochPath = path + "/epoch";
+      if (!BasicValidator.validateString(epochPath, "epoch", epoch, context)) {
+        result = false;
+      } else {
+        const epochPattern = /^[0-9]+(\.[0-9]+)?$/;
+        if (!epochPattern.test(epoch)) {
+          const message =
+            "The 'epoch' must match the pattern '^[0-9]+(\\.[0-9]+)?$'";
+          const issue = JsonValidationIssues.STRING_VALUE_INVALID(
```

Comment:
Admittedly, there is some overlap between the meaning of `STRING_VALUE_INVALID` and `STRING_PATTERN_MISMATCH`. I think both are OK here.